### PR TITLE
Upgrade test improvements

### PR DESCRIPTION
**Pull request by sabaini**
# Description
Test snap upgrades for reef/stable. Also add some verification

## Type of change

- [x ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

This PR is a test fix

## Contributor's Checklist

Please check that you have:

- [x] self-reviewed the code in this PR.
- [x] added code comments, particularly in hard-to-understand areas.

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -382,16 +382,25 @@ jobs:
...
           juju wait-for application microceph --query='forEach(units, unit => unit.workload-status=="active" && unit.agent-status=="idle")' --timeout=30m || true
           # Get status
           juju ssh microceph/0 -- sudo ceph -s
+          juju ssh microceph/0 -- sudo microceph cluster list
           juju status
           # Fail if goal status not reached
           juju wait-for application microceph --query='forEach(units, unit => unit.workload-status=="active" && unit.agent-status=="idle")' --timeout=2m
           date
+          # Test if we have expected snap releases
+          juju exec -a microceph -- sudo snap info microceph
```
Comment (lmlg) [inline .github/workflows/build-and-test.yml:398]: Is this line needed?

**Review by lmlg — APPROVED** [review]
lgtm, just a small question

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (sabaini) [reply .github/workflows/build-and-test.yml:398]: This was for informational purposes -- i.e. if the test in the next line fails we could scroll back and see which snap versions were actually active

Comment (lmlg) [reply .github/workflows/build-and-test.yml:398]: Figured that was the case.

**Pull request merged**